client/core/trade.go

```diff
@@ -2136,6 +2158,70 @@ func (t *trackedTrade) auditContract(match *matchTracker, coinID, contract, txDa
 	if err != nil {
 		return err
 	}
+
+	t.mtx.Lock()
+	defer t.mtx.Unlock()
+
+	err = t.validateAuditInfo(auditInfo, match, coinID, contract)
+	if err != nil {
+		return err
+	}
+	// Audit successful. Update status and other match data.
+	proof := &match.MetaData.Proof
+	if match.Side == order.Maker {
+		match.Status = order.TakerSwapCast
+		proof.TakerSwap = coinID
+	} else {
+		proof.SecretHash = auditInfo.SecretHash
+		match.Status = order.MakerSwapCast
+		proof.MakerSwap = coinID
+	}
+	proof.CounterTxData = txData
+	proof.CounterContract = contract
+	match.counterSwap = auditInfo
+
+	err = t.db.UpdateMatch(&match.MetaMatch)
+	if err != nil {
+		t.dc.log.Errorf("Error updating database for match %v: %s", match, err)
+	}
+
+	t.dc.log.Infof("Audited contract (%s: %v) paying to %s for order %s, match %s, with tx data = %t",
+		t.wallets.toAsset.Symbol, auditInfo.Coin, auditInfo.Recipient, t.ID(), match, len(txData) > 0)
+
+	return nil
+}
+
+// reAuditContract performs a repeat contract audit for the specified match and
+// returns an error if the repeat audit fails. This does NOT update match fields
+// but the trackedTrade mtx MUST be locked for reading mutable match fields.
+func (t *trackedTrade) reAuditContract(match *matchTracker) error {
+	if match.Side == order.Maker && match.Status != order.TakerSwapCast {
+		return fmt.Errorf("Invalid repeat audit request for maker at status %s.", match.Status)
+	}
+	if match.Side == order.Taker && match.Status != order.MakerSwapCast {
+		return fmt.Errorf("Invalid repeat audit request for taker at status %s.", match.Status)
+	}
+
+	proof := &match.MetaData.Proof
+	coinID, contract := match.counterSwap.Coin.ID(), proof.CounterContract
+	auditInfo, err := t.wallets.toWallet.AuditContract(coinID, contract, nil, time.Time{})
```

Comment:
Does `matchTime` not matter here because we know the `externalTx` is already cached? Shouldn't we just pass the actual `match.matchTime()` here anyway?

Reply:
🤔